connectors/sources/mongo.py

```diff
@@ -96,38 +97,36 @@ class MongoDataSource(BaseDataSource):
     def __init__(self, configuration):
         super().__init__(configuration=configuration)
 
-        client_params = {}
+        self.client_params = {}
+        self.db = None
+        self.collection = None
+        self.client = None
+        self.host = self.configuration["host"]
 
-        host = self.configuration["host"]
         user = self.configuration["user"]
         password = self.configuration["password"]
         ssl_ca = self.configuration["ssl_ca"]
         tls_insecure = self.configuration["tls_insecure"]
         self.certfile = ""
 
         if self.configuration["direct_connection"]:
-            client_params["directConnection"] = True
+            self.client_params["directConnection"] = True
 
         if len(user) > 0 or len(password) > 0:
-            client_params["username"] = user
-            client_params["password"] = password
+            self.client_params["username"] = user
+            self.client_params["password"] = password
 
         if self.configuration["ssl_enabled"]:
-            client_params["tls"] = True
+            self.client_params["tls"] = True
             if ssl_ca:
                 pem_certificates = get_pem_format(key=ssl_ca)
                 with NamedTemporaryFile(mode="w", suffix=".pem", delete=False) as cert:
                     cert.write(pem_certificates)
                     self.certfile = cert.name
-                client_params["tlsCAFile"] = self.certfile
-            client_params["tlsInsecure"] = tls_insecure
+                self.client_params["tlsCAFile"] = self.certfile
+            self.client_params["tlsInsecure"] = tls_insecure
         else:
-            client_params["tls"] = False
-
-        self.client = AsyncIOMotorClient(host, **client_params)
-
-        self.db = self.client[self.configuration["database"]]
-        self.collection = self.db[self.configuration["collection"]]
+            self.client_params["tls"] = False
```

Comment:
This code can go to `_get_client`

Reply:
As we have moved this code to `get_client`. I have added a code for removing the temp file in the `finally` block. So, every time it deletes the file when context is over so i guess its fix this [issue](https://github.com/elastic/connectors/issues/1948) as well